hack/ci-e2e.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+set -x
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}" || exit 1
+
+# Ensure requirements are installed
+"${REPO_ROOT}/hack/e2e/ensure_go.sh"
+export PATH=$PATH:/usr/local/go/bin
+"${REPO_ROOT}/hack/e2e/ensure_minikube.sh"
+"${REPO_ROOT}/hack/e2e/ensure_kubectl.sh"
+"${REPO_ROOT}/hack/e2e/ensure_cmctl.sh"
```

Comment:
I see both you and @mboukhalfa asked about this so I will answer both here.
In an earlier commit I had some comments to explain it but I unfortunately squashed a bit too much so they are now gone.

We can definitely use `kubectl` to install cert-manager. The reason I used `cmctl` here is because it also waits for cert-manager to be completely available. It even tests this by creating a test issuer and certificate. Actually `clusterctl init` does the same, which is how we install cert-manager most of the time I think.
My reasoning for use cmctl was that I did not want to write the code to wait and test cert-manager right now and make the PR bigger. In the long run I expect that we will not use `cmctl` nor `kubectl` for any of this. Instead we should use native go code, but for now I think this is the easiest way to get a reliable installation of cert-manager.

Reply:
Updated the PR description with a Q/A for `cmctl` now. Also added a couple of points under Future work about the possibility to get rid of `os.Exec` and use native go code instead.

Reply:
Thanks, I have looked at it before, but it would basically double the size of the code in this PR. It is several hundred lines of code that we would need to copy/adapt from CAPI.

Unfortunately they did not expose this part of the API in a way that makes it easy for us to use. The [newCertManagerClient](https://github.com/kubernetes-sigs/cluster-api/blob/a36712e28bf5d54e398ea84cb3e20102c0499426/cmd/clusterctl/client/cluster/cert_manager.go#L96C12-L96C12) function is private so we cannot use it. Even if it was public it would be questionable to use it in BMO context directly because it is tied to CAPI specific things like the `repositoryClientFactory`.
There is a publicly exposed way to get a CertManagerClient through the [cluster.Client](https://github.com/kubernetes-sigs/cluster-api/blob/a36712e28bf5d54e398ea84cb3e20102c0499426/cmd/clusterctl/client/cluster/client.go#L47) interface. But this is again tied to everything from workload clusters to providers so I did not look at it more for now.